### PR TITLE
Support fork id upgrades without down time for the trusted state

### PR DESCRIPTION
**Pull request by agnusmor**
Closes [#2236](https://github.com/0xPolygonHermez/zkevm-node/issues/2236).

### What does this PR do?

Support fork id upgrades without down time for the trusted state.

Steps to perform a forkid upgrade using this feature (e.g upgrading from forkid 4 to 5):
1. Update node/executor to new forkid 5 versions. Keep provers running on previous version (previous forkid 4)
2. Set ForkUpgradeBatchNumber config parameter to the batch number that will be the last batch using the previous forkid 4 and set ForkUpgradeNewForkId to the new forkid value (5). 
3. Start/restart node components to get the new config values 
4. Wait until ForkUpgradeBatchNumber batch has been virtualized and verfied. While waiting for this the system we will continue processing txs and batches and adding it to the trusted state (but virtualization and verification will be on hold from ForkUpgradeNewForkId+1 batch)
5. Update SC to the new forkid 5
6. Stop provers using previous forkid 4
7. Set ForkUpgradeNewForkId = 0 and ForkUpgradeNewForkId = 0 to disable this feature
8. Restart node components (RPC, Synchornizer, Sequencer, SequenceSender, Aggregator) to get the config changes
9. Update provers to new forkid 5 versions
10. Check the service resumes correctly and new batches are virtualized and verified using new forkid 5

New config parameters in the node (at root level of the toml file):
ForkUpgradeBatchNumber = 0
ForkUpgradeNewForkId = 0

### Reviewers

Main reviewers:
@ToniRamirezM 
@ARR552

### CODE REVIEW

File: test/config/debug.node.config.toml [hunk 1]
```diff
@@ -105,8 +105,7 @@ PrivateKeys = [{Path = "./test/sequencer.keystore", Password = "testonly"}]
 
 [Aggregator]
 Host = "0.0.0.0"
-Port = 50081
-ForkId = 2
+Port = 500
```
Comment (ARR552) [inline test/config/debug.node.config.toml:108]: Why this change? I think it is a typo error

File: aggregator/config.go [hunk 2]
```diff
@@ -60,7 +60,7 @@ type Config struct {
 	ChainID uint64
 
 	// ForkID is the L2 ForkID provided by the Network Config
-	ForkId uint64 `mapstructure:"ForkId"`
+	ForkId uint64
```
Comment (ARR552) [inline aggregator/config.go:63]: How do you assign this variable? If it is not needed you can delete it

File: test/config/debug.node.config.toml (hunk 1 above)
Comment (agnusmor) [reply test/config/debug.node.config.toml:108]: Yes, is a typo, only Forkid must be deleted

File: aggregator/config.go (hunk 2 above)
Comment (agnusmor) [reply aggregator/config.go:63]: It is assigned in run.go, after read the forkid intervals from L1 (ChanID is also assigned in this way):
https://github.com/0xPolygonHermez/zkevm-node/blob/b801f527572adb919ebf3d99be8bc603ddc24a70/cmd/run.go#L132

**Review by ToniRamirezM — APPROVED** [review]

**Review by ARR552 — APPROVED** [review]
